contracts/StablecoinConverter.sol

```diff
@@ -224,23 +226,81 @@ contract StablecoinConverter is EpochTokenLocker {
         tokenConservation.checkTokenConservation();
         documentTrades(batchIndex, owners, orderIds, volumes, tokenIdsForPrice);
     }
+    /**
+     * Public View Methods
+     */
+
+    /** @dev View returning ID of listed tokens
+      * @param addr address of listed token.
+      * @return tokenId as stored, via BiMap, within the contract.
+      */
+    function tokenAddressToIdMap(address addr) public view returns (uint16) {
+        return IdToAddressBiMap.getId(registeredTokens, addr);
+    }
+
+    /** @dev View returning address of listed token by ID
+      * @param id tokenId as stored, via BiMap, within the contract.
+      * @return address of (listed) token
+      */
+    function tokenIdToAddressMap(uint16 id) public view returns (address) {
+        return IdToAddressBiMap.getAddressAt(registeredTokens, id);
+    }
 
+    /** @dev View returning all currently stored, byte-encoded sell orders
+      * @return encoded bytes representing all orders ordered by (user, index)
+      */
+    function getEncodedAuctionElements() public view returns (bytes memory elements) {
+        if (allUsers.size() > 0) {
+            address user = allUsers.first();
+            bool stop = false;
+            while (!stop) {
+                for (uint i = 0; i < orders[user].length; i++) {
+                    Order memory order = orders[user][i];
+                    elements = elements.concat(encodeAuctionElement(
+                        user,
+                        getBalance(user, tokenIdToAddressMap(order.sellToken)),
+                        order
+                    ));
+                }
+                if (user == allUsers.last) {
+                    stop = true;
+                } else {
+                    user = allUsers.next(user);
+                }
+            }
+        }
+        return elements;
+    }
+
+    /** @dev called by solvers to determine their solution better than the current winner.
+      * @return objective function evaluation of the currently winning solution, or zero if no solution proposed.
+      */
     function getCurrentObjectiveValue() public view returns(uint) {
         if (previousSolution.batchId == getCurrentBatchId() - 1) {
             return previousSolution.objectiveValue;
         } else {
             return 0;
         }
     }
+    /**
+     * Internal Functions
+     */
 
+    /** @dev called at the end of submitSolution with a value of tokenConservation / 2
+      * @param feeReward amount to be rewarded to the solver
+      */
     function grantRewardToSolutionSubmitter(uint feeReward) internal {
         previousSolution.feeReward = feeReward;
         addBalanceAndBlockWithdrawForThisBatch(msg.sender, tokenIdToAddressMap(0), feeReward);
     }
 
+    /** @dev called early from within submitSolution to update the token prices.
```

Comment:
What does "early" mean? Can we be more specific. Also I'm not sure if we need to document all internal methods as well?

Reply:
I would be happy to remove the internal method documentation. Does anyone else have any opinion here? @josojo, @nlordell?

Reply:
Personally, I don't think they are needed but there is also no harm in including them.